### PR TITLE
feat: sign images using cosign

### PR DESCRIPTION
**Pull request by JamieMagee**
Signs all images automatically using cosign keyless workflow. Tested on my own fork of `containerbase/python`[^1] (The build log contains some additional logging not present in this PR)

This requires cosign to be installed in all workflows where container images are published, but the sigstore team provides an action to do this[^2], and I've opened an issue to have it be included in GitHub Actions workers by default [^3]

Closes https://github.com/containerbase/buildpack/issues/201

References:
- https://github.blog/2021-12-06-safeguard-container-signing-capability-actions/
- https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml

Related PRs:
- https://github.com/containerbase/java/pull/3
- https://github.com/containerbase/node/pull/6
- https://github.com/containerbase/php/pull/3
- https://github.com/containerbase/python/pull/3
- https://github.com/containerbase/ruby/pull/5
- https://github.com/containerbase/sidecar/pull/3
- https://github.com/renovatebot/docker-cocoapods/pull/52
- https://github.com/renovatebot/docker-dotnet/pull/44
- https://github.com/renovatebot/docker-elixir/pull/41
- https://github.com/renovatebot/docker-go/pull/96
- https://github.com/renovatebot/docker-helm/pull/8
- https://github.com/renovatebot/docker-java/pull/70
- https://github.com/renovatebot/docker-node/pull/137
- https://github.com/renovatebot/docker-php/pull/52
- https://github.com/renovatebot/docker-python/pull/69
- https://github.com/renovatebot/docker-renovate-full/pull/392
- https://github.com/renovatebot/docker-renovate/pull/243
- https://github.com/renovatebot/docker-ruby/pull/60
- https://github.com/renovatebot/docker-rust/pull/51
- https://github.com/renovatebot/docker-sidecar/pull/3
- https://github.com/renovatebot/docker-swift/pull/35

[^1]: https://github.com/JamieMagee/python/runs/4646250456?check_suite_focus=true
[^2]: https://github.com/sigstore/cosign-installer
[^3]: https://github.com/actions/virtual-environments/issues/4813

### CODE REVIEW

File: src/utils/docker.ts [hunk 1]
```diff
@@ -231,6 +233,17 @@ export async function publish({
     log.warn(chalk.yellow('[DRY_RUN]'), chalk.blue('Would push:'), fullName);
   } else {
     await docker('push', fullName);
+    await sign(fullName);
```
Comment (viceice) [inline src/utils/docker.ts:236]: I think this should be configurable

Comment (JamieMagee) [reply src/utils/docker.ts:236]: Why would we _not_ sign an image?

Comment (viceice) [reply src/utils/docker.ts:236]: 🤷‍♂️🤔

File: src/utils/cosign/common.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { exec } from '../../util';
+import { ExecResult } from '../types';
```
Comment (viceice) [inline src/utils/cosign/common.ts:2]:
```suggestion
import type { ExecResult } from '../types';
```

File: test/utils/cosign/common.spec.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import { mocked } from '../../utils';
+
+jest.mock('../../../src/util');
+
+const utils = mocked(_utils);
+
+describe('utils/cosign', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
```
Comment (viceice) [inline test/utils/cosign/common.spec.ts:12]: I don't think we need this. 😉

**Review by viceice — APPROVED** [review]

**Pull request merged**